### PR TITLE
Fix singleplayer crashes

### PR DESCRIPTION
**Pull request by z33ky**
I encountered and debugged these two crashes playing the Jedi Outcast singleplayer on Linux.
I've also applied the codeJK2 fix to code(JK3) since I would bet it can happen there as well.

### CODE REVIEW

File: code/cgame/cg_credits.cpp [hunk 1]
```diff
@@ -442,7 +437,7 @@ void CG_Credits_Init( const char *psStripReference, vec4_t *pv4Color)
 					{
 						// sort entries RHS dotted entries by alpha...
 						//
-						qsort(&CreditLine.vstrText[0], CreditLine.vstrText.size(), sizeof(CreditLine.vstrText[0]), SortBySurname);
+						std::sort( CreditLine.vstrText.begin(), CreditLine.vstrText.end(), SortBySurname );
```
Comment (z33ky) [inline code/cgame/cg_credits.cpp:440]: Compared to the previous code, this will additionally require the header <algorithm>.
It seems headers are mostly not included in this implementation file (i.e. <string> or <vector> isn't included directly). It compiled on my machine, so I left the includes as-is. Let me know if I need to add that somewhere to accommodate this change.

File: code/rd-vanilla/tr_surface.cpp [hunk 2]
```diff
@@ -1617,6 +1617,10 @@ static void RB_SurfaceLathe()
 	{
 		lod = 4;
 	}
+	if ( lod < 1 )
+	{
+		lod = 1;
+	}
```
Comment (xycaleth) [inline code/rd-vanilla/tr_surface.cpp:1623]: What does this change? Seems unrelated to the other changes :)

Comment (xycaleth) [reply code/rd-vanilla/tr_surface.cpp:1623]: Actually never mind, I see the problem

**Review by xycaleth — APPROVED** [review]

**Comment by xycaleth** [comment]
Thanks for the contribution @z33ky!
